project/src/components/login-form/login-form.tsx

```diff
@@ -0,0 +1,123 @@
+import styles from './login.module.css';
+import { ChangeEvent, FormEvent, useState } from 'react';
+import { useAppDispatch } from '../../hooks';
+import { EMAIL_REGEXP, PASSWORD_REGEXP, LOGIN_FIELDS } from '../../const';
+import { AuthData, loginAction } from '../../store/slices/user-slice';
+import cn from 'classnames';
+
+type Field = {
+  value: string;
+  regex: RegExp;
+  error: boolean;
+  errorText: string;
+}
+
+const LoginForm = () => {
+  const [formData, setFormData] = useState<Record<string, Field>>({
+    email: {
+      value: '',
+      error: false,
+      regex: EMAIL_REGEXP,
+      errorText: 'Email is incorrect'
+    },
+    password: {
+      value: '',
+      error: false,
+      regex: PASSWORD_REGEXP,
+      errorText: 'Password should contain at least 1 letter and 1 number'
+    },
+  });
+
+  const dispatch = useAppDispatch();
+
+  const isFieldsGroupValid = () => formData.email.error || formData.password.error;
+
+  const handleInputChange = (e: ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = e.target;
+
+    setFormData({
+      ...formData,
+      [name]: {
+        ...formData[name],
+        value: value,
+      }
+    });
+
+    const isValid = (inputValue: string, matchedValue: RegExp) => matchedValue.test(inputValue);
+
+    if (!isValid(value, formData[name].regex)) {
+      setFormData({
+        ...formData,
+        [name]: {
+          ...formData[name],
+          error: true,
+          value: value
+        }
+      });
+    } else {
+      setFormData({
+        ...formData,
+        [name]: {
+          ...formData[name],
+          error: false,
+          value: value},
+      });
+    }
+  };
+
+  const onSubmit = (authData: AuthData) => {
+    dispatch(loginAction(authData));
+  };
+
+  const handleSubmit = (evt: FormEvent<HTMLFormElement>) => {
+    evt.preventDefault();
+    onSubmit({
+      login: formData.email.value,
+      password: formData.password.value,
+    });
+  };
+
+  return (
+    <form
+      className="login__form form"
+      action=""
+      onSubmit={handleSubmit}
+      method="post"
+    >
+
+      {
+        Object.entries(LOGIN_FIELDS).map(([name, label]) => (
+          <div key={name} className="login__input-wrapper form__input-wrapper">
+            <label className="visually-hidden">{label}</label>
+            <div className={styles.error}>
+              {
+                formData[name].error && (
+                  <div className={styles.error}>{formData[name].errorText}</div>
+                )
+              }
```

Comment:
я бы вот так отформатировал бы, просто для инфо
```suggestion
              {formData[name].error && (
                  <div className={styles.error}>{formData[name].errorText}</div>
              )}
```